lib/http_proxy_override.dart

```diff
@@ -14,22 +14,69 @@ Future<String?> _getProxyPort() async {
 }
 
 class HttpProxyOverride extends HttpOverrides {
-  late final String? host;
-  late final String? port;
+  /// The host part of the proxy address.
+  final String? host;
 
-  HttpProxyOverride._(this.host, this.port);
+  /// The port part of the proxy address.
+  final String? port;
 
-  static Future<HttpProxyOverride> createHttpProxy() async {
-    return HttpProxyOverride._(await _getProxyHost(), await _getProxyPort());
+  /// Configures whether a secure connection to a host should be allowed with
+  /// a server certificate that cannot be authenticated by any of the trusted
+  /// root certificates.
+  final bool ignoreBadCertificates;
+
+  final SecurityContext securityContext;
+
+  HttpProxyOverride._(
+      this.host, this.port, this.ignoreBadCertificates, this.securityContext);
+
+  /// Create an instance of [HttpProxyOverride].
+  ///
+  /// Reads the configured proxy host and port from the underlying platform
+  /// and configures the [HttpClient] to use the proxy. The proxy settings are
+  /// read once at creation time.
+  ///
+  /// [ignoreBadCertificates] configures whether a secure connection to a host
+  /// should be allowed with a server certificate that cannot be authenticated
+  /// by any of our trusted root certificates.
+  /// For example, this can be useful when using debugging proxies like Charles
+  /// or mitmproxy during development.
+  /// **Do not enable this in production unless you are 100% sure.** Setting
+  /// this enables MITM attacks.
+  /// Default: `false`.
+  ///
+  /// With [securityContext] a [SecurityContext] can be provided that is used to
+  /// construct the [HttpClient]. This can be useful to provide a
+  /// [SecurityContext] that is configured with certificates that a proxy
+  /// server requires.
+  ///
+  /// Supported platforms to read proxy settings from are **iOS** and
+  /// **Android**.
+  static Future<HttpProxyOverride> create(
+      {bool ignoreBadCertificates = false,
+      SecurityContext? securityContext}) async {
+    return HttpProxyOverride._(
+        await _getProxyHost(),
+        await _getProxyPort(),
+        ignoreBadCertificates,
+        securityContext ?? SecurityContext.defaultContext);
   }
 
   @override
   HttpClient createHttpClient(SecurityContext? context) {
+    if (context == null) {
```

Comment:
The `SecurityContext` should be handled when the `HttpClient` is created, see https://api.dart.dev/stable/2.14.4/dart-io/HttpClient/HttpClient.html, but not set by this library, say that the user should 
```dart
  final context = SecurityContext.defaultContext;
  context.setTrustedCertificatesBytes(...);
  HttpClient httpClient = HttpClient(context: context);

  HttpProxyOverride httpProxyOverride = await HttpProxyOverride.create();
```

but not
```dart
  HttpClient httpClient = HttpClient();

  final context = SecurityContext.defaultContext;
  context.setTrustedCertificatesBytes(...);
  HttpProxyOverride httpProxyOverride = await HttpProxyOverride.create(securityContext: context);
```